lib/rgeo/shapefile/reader.rb

```diff
@@ -176,6 +176,14 @@ def self.open(path_, opts_ = {}, &block_)
       # block. You should use Reader::open instead.
 
       def initialize(path_, opts_ = {}) # :nodoc:
+        @allow_unsafe = false
+        if opts_[:unsafe_methods]
+          if Gem::Version.new(RGeo::VERSION) < Gem::Version.new('3.0.0')
+            warn "RGeo v#{RGeo::VERSION} does not support unsafe methods. Unsafe methods were added in RGeo v3.0.0."
```

Comment:
Took a swing at a reasonable warning - I honestly haven't tested this but I can if you'd like

Reply:
Okay I've tested this now and it works